pkg/network/tracer/tracer.go

```diff
@@ -394,46 +403,73 @@ func (t *Tracer) Stop() {
 }
 
 // GetActiveConnections returns the delta for connection info from the last time it was called with the same clientID
-func (t *Tracer) GetActiveConnections(clientID string) (*network.Connections, error) {
+func (t *Tracer) GetActiveConnections(clientID string, maxConnectionPerMessage int) (*network.Connections, bool, error) {
```

Comment:
I think it would be better if the paging was one level up as it is more a function of the request. There is also this for reference: https://cloud.google.com/apis/design/design_patterns#list_pagination

Reply:
I can't do one level up, as the goal is to reduce the allocation during the aggregation of dns, http by processing only a page/subset of connections. NPM connections as reference.

On paging we can embedded the mechanism in the protobuf, but would matter as it's used only on our internal api (unix socket)

Reply:
The paging needs to be a little more sophisticated to handle cases like where process-agent may crash/restart. Since process-agent uses the same client id every time, just using client id to track pages won't be sufficient.

For moving one level up, I meant do the pagination in modules/network_tracer.go. Another option would be to add something like `GetActiveConnectionsPaged` to the tracer which would entail fewer side-affects to existing code.

Reply:
Adding `GetActiveConnectionsPaged` sound good
moving up to modules/network_tracer.go it's too late we will got the allocation spike from dns, http already

Reply:
I thought the allocations were coming from aggregations, and those happen in `writeConnections`, right? In the call to `Marshal`?

Reply:
yes it happen on writeConnections, that why I give only a [subset of connections](https://github.com/DataDog/datadog-agent/pull/16806/files#diff-d55acaf7f363a35345fe9cceca15bb12a45cbf1b31d05a91853f4d285fb6361aR517) 

Next patch would fix the GetDelta()->GetStats per protocols that iterate only on a subset of connections, this would limit the allocation even more